File: DAC_API/Models/DTO/UserAchievement/UserAchievementResponseDTO.cs
using System;

namespace DAC_API.Models.DTO.UserAchievement {
    public class UserAchievementResponseDTO {
        public int UserId { get; set; }
        public int AchievementId { get; set; }
        public DateTime? EarnedAt { get; set; }

        public string Username { get; set; }
        public string AchievementName { get; set; }
        public string AchievementRequirements { get; set; }
    }
}

File: DailyAdventureChallenge/app/src/main/java/com/example/dailyadventurechallenge/data/dto/Achievement/UserAchievementResponseDTO.kt
package com.example.dailyadventurechallenge.data.dto.Achievement

import com.google.gson.annotations.SerializedName

data class UserAchievementResponseDTO(
    @SerializedName("userId")
    val userId: Int,

    @SerializedName("achievementId")
    val achievementId: Int,

    @SerializedName("earnedAt")
    val earnedAt: String,

    @SerializedName("username")
    val username: String,

    @SerializedName("achievementName")
    val achievementName: String,

    @SerializedName("achievementRequirements")
    val achievementRequirements: String
)
